apis/v1alpha2/shared_types.go

```diff
@@ -383,3 +383,58 @@ type AnnotationKey string
 // +kubebuilder:validation:MinLength=0
 // +kubebuilder:validation:MaxLength=4096
 type AnnotationValue string
+
+// AddressRouteMatches defines AddressMatch rules for inbound traffic according to
+// source and/or destination address of a packet or connection.
+type AddressRouteMatches struct {
+	// SourceAddresses indicates the originating (source) network
+	// addresses which are valid for routing traffic.
+	//
+	// Support: Core
+	SourceAddresses []AddressMatch `json:"sourceAddresses"`
+
+	// DestinationAddresses indicates the destination network addresses
+	// which are valid for routing traffic.
+	//
+	// Support: Core
+	DestinationAddresses []AddressMatch `json:"destinationAddresses"`
+}
+
+// AddressMatch defines matching rules for network addresses by type.
+type AddressMatch struct {
```

Comment:
What's the motivation for a new type that is practically the same as `GatewayAddress`? Is this still needed after the refactoring?

Reply:
We don't accept `Hostname` for L4 matching.

Reply:
Hmm, is that tradeoff worth it? We could reject `Hostname` in the webhook and not bifurcate the types ..

Reply:
This is now re-converged and there's no longer a new type.